src/screens/Filter/components/FilterItem/useFilterItemController.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from 'react';
+
+import { FilterItemValueModel } from 'src/models';
+
+export interface UseFilterItemControllerParams {
+  filterTitle: string;
+  filterName: string;
+  values: Array<FilterItemValueModel>;
+  multiple: boolean;
+  onFilterChange: (filterName: string, value: string) => void;
+}
+
+export const useFilterItemController = (params: UseFilterItemControllerParams) => {
+
+  const onMultipleFilterChange = useCallback((value: string, previousValueString: string) => {
+    if (previousValueString.includes(value)) {
+
+      return params.onFilterChange(params.filterName, previousValueString.split(',')
+        .filter((item) => item !== value).toString().replaceAll(',', '|'));
+    }
+
+    const currentValuesString = previousValueString ? previousValueString.replaceAll(',', '|') + `|${value}` : value;
+
+    return params.onFilterChange(params.filterName, currentValuesString);
+  }, [params.multiple]);
+
+  const handleChange = useCallback((value: string) => {
+    const previousValueString = params.values.filter((v) => v.isActive === true)
+      .map((item) => item.value)
+      .toString();
```

Comment:
do not use toString for arrays as it leads to confusion and increases the cognitive complexity of your code because depending on the array content, the result will be different
<img width="166" alt="image" src="https://github.com/valeskus/RecipeApp/assets/25794983/f5f2c22a-57b9-4b14-94e9-730f97197bf1">